### PR TITLE
Made fill_value a property of file handle

### LINKED ISSUE #146: Make fill_value a property of the file handle
There would seem to be a great benefit of `fh.read(...)` to have only arguments that are valid for all formats. We've gone partially that way by moving `squeeze` up and I think we should continue that by also moving `fill_value` to be an argument of `open` rather than of `read` (this in particular since `gsb` does not have a fill value; #105). I think initially at least I would advocate for not making this settable after opening the file (though that is not hard to do in principle; it would require invalidating any frame already decoded, though).

### PR DESCRIPTION
**Pull request by theXYZT**
Fixes #146

### CODE REVIEW

**Comment by theXYZT** [comment]
Removed `fill_value` from GSB Readers. This should deal with #105.

File: baseband/vdif/base.py [hunk 1]
```diff
@@ -554,14 +555,16 @@ def write(self, data, invalid_data=False):
...
+    Value to use for invalid or missing data. Default: 0.
 squeeze : bool, optional
     If `True` (default), remove any dimensions of length unity from
     decoded data.
 
 --- For writing : (see :class:`VDIFStreamWriter`)
 
 nthread : int
-    Number of threads the VLBI data has (e.g., 2 for 2 polarisations).
+    Number of threads the VLBI data has (e.g., 2 for 2 polarizations).
```
Comment (mhvk) [inline baseband/vdif/base.py]: Hey, why the american spelling?

File: baseband/vlbi_base/base.py [hunk 2]
```diff
@@ -79,6 +81,11 @@ def __init__(self, fh_raw, header0, unsliced_shape, bps, complex_data,
         self._subset = subset
         self._sample_shape = self._get_sample_shape(subset_wrapints)
 
+    @property
```
Comment (mhvk) [inline baseband/vlbi_base/base.py]: Even though the private `_fill_value` attribute is set in the base, I think it makes sense to only expose it in the reader; for the writer at least currently it is unused so would only lead to confusion. (I'm wondering whether I can do the hierarchy better such that the writer doesn't even have the attribute, but that's definitely beyond the scope of this PR!)

**Comment by mhvk** [comment]
p.s.  I can do the rebasing if you want.

**Comment by cczhu** [comment]
Ahh, I just changed master, so before merging, you'll need to rebase off of upstream master:
```
git fetch --all
git rebase upstream/master
(if the previous step is successful)
git push -f origin master
```

**Comment by mhvk** [comment]
@theXYZT - I did the rebase/squash for you. One aside: in general, it is better practice to do work in a named branch on your fork of the repository; that way your own master branch has a known state.

**Pull request merged**